src/ark/utils/data_utils.py

```diff
@@ -54,119 +56,193 @@ def save_fov_mask(fov, data_dir, mask_data, sub_dir=None, name_suffix=''):
     image_utils.save_image(os.path.join(save_dir, fov_file), mask_data)
 
 
-def relabel_segmentation(labeled_image, labels_dict):
-    """Takes a labeled image and translates its labels according to a dictionary.
-
-    Returns the relabeled array (according to the dictionary).
-
-    Args:
-        labeled_image (numpy.ndarray):
-            2D numpy array of labeled cell objects.
-        labels_dict (dict):
-            a mapping between labeled cells and their clusters.
-
-    Returns:
-        numpy.ndarray:
-            The relabeled array.
+class CellClusterMaskData:
+    """
+    A class containing the cell labels, cluster labels, and segmentation labels for the
+    whole cohort. Also contains the mapping from the segmentation label to the cluster
+    label for each FOV.
     """
 
-    img = np.copy(labeled_image)
-    unique_cell_ids = np.unique(labeled_image)
-    unique_cell_ids = unique_cell_ids[np.nonzero(unique_cell_ids)]
-
-    default_label = max(labels_dict.values()) + 1
-
-    # cast to int16 to allow for Photoshop loading
-    relabeled_img = np.vectorize(
-        lambda x: labels_dict.get(x, default_label) if x != 0 else 0
-    )(img).astype('int16')
-
-    return relabeled_img
-
-
-def label_cells_by_cluster(fov, all_data, label_map, fov_col=settings.FOV_ID,
-                           cell_label_column=settings.CELL_LABEL,
-                           cluster_column=settings.KMEANS_CLUSTER):
-    """Translates cell-ID labeled images according to the clustering assignment.
+    fov_column: str
+    label_column: str
+    cluster_column: str
+    unique_fovs: List[str]
+    unassigned_id: int
+    mapping: DataFrameGroupBy
+
+    def __init__(self, data: pd.DataFrame, fov_col: str, label_col: str, cluster_col: str) -> None:
+        """
+        A class containing the cell data, cell label column, cluster column and the mapping from a
+        cell label to a cluster.
+
+        Args:
+            data (pd.DataFrame):
+                A cell table with the cell label column and the cluster column.
+            fov_col (str):
+                The name of the column in the cell table that contains the FOV ID.
+            label_col (str):
+                The name of the column in the cell table that contains the cell label.
+            cluster_col (str):
+                The name of the column in the cell table that contains the cluster label.
+        """
+        self.fov_column: str = fov_col
+        self.label_column: str = label_col
+        self.cluster_column: str = cluster_col
+
+        # Extract only the necessary columns: fov ID, segmentation label, cluster label
+        mapping_data: pd.DataFrame = data[[self.fov_column,
+                                           self.label_column, self.cluster_column]].copy()
+
+        mapping_data[self.cluster_column] = mapping_data[self.cluster_column].astype(np.int64)
+        mapping_data[self.label_column] = mapping_data[self.label_column].astype(np.int64)
```

Comment:
I guess not super important, but related to Cami's comment, np.int32 would probably be okay here?